### PR TITLE
new example displaying entry info with kanji and reading

### PR DESCRIPTION
**Pull request by ultrasaurus**
Cool library!  I'm not a Rust expert, so this might not be the best example code, but I had a hard time figuring out how to generate this kind of info from the reference documentation.  I thought it might be helpful to others if you wanted to merge it.  

Would love feedback and happy to improve this, or, of course, feel free to modify if you are inspired to provide a better example along these lines.  (I'm still trying to wrap my head around what information is available in the dictionary and how it is structured, as well as learning the Rust language!)

### CODE REVIEW

File: examples/kanji_info.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+fn main() {
+    let input = "一日";
+    let count = jmdict::entries()
+        .filter(|e| {
+            let mut result = false;
+            if e.kanji_elements().any(|k| k.text == input) {
+                result = true;
+                println!("--- entry #{} ---", e.number);
+                let kanji_elements: Vec<KanjiElement> = e.kanji_elements().map(|ke| ke).collect();
+                for kanji in &kanji_elements {
```
Comment (majewsky) [inline examples/kanji_info.rs]: The magic of rust-jmdict's iterator-based interface is that you don't need to materialize into a Vec most of the time. You can just do:
```suggestion
                for kanji in e.kanji_elements() {
```
without any extra allocations. Same for `e.reading_elements()` below.

File: examples/kanji_info.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+*******************************************************************************/
+use jmdict::{KanjiElement, ReadingElement};
+
+fn main() {
+    let input = "一日";
+    let count = jmdict::entries()
+        .filter(|e| {
+            let mut result = false;
+            if e.kanji_elements().any(|k| k.text == input) {
+                result = true;
```
Comment (majewsky) [inline examples/kanji_info.rs]: You can avoid carrying the `result` variable through the entire function, and also avoid an extra level of indentation, by inverting the check and exiting early on failure:
```rust
            if e.kanji_elements().all(|k| k.text != input) {
                return false;
            }
            // ...
            true
```

**Comment by ultrasaurus** [comment]
I believe I've addressed all of the suggested improvements -- thanks so much for your feedback!

**Comment by majewsky** [comment]
I've applied another simplification to this example, using the `.inspect()` method that I've become aware of recently: https://github.com/majewsky/rust-jmdict/commit/e26e3b05cb0514687f37d6d9e10b0605d0924b5c

**Pull request merged**